core/src/main/java/org/springframework/security/scheduling/DelegatingSecurityContextTaskScheduler.java

```diff
@@ -30,47 +35,64 @@
  * @author Richard Valdivieso
  * @since 5.1
  */
-public class DelegatingSecurityContextTaskScheduler implements TaskScheduler {
+public class DelegatingSecurityContextTaskScheduler extends AbstractDelegatingSecurityContextSupport
```

Comment:
Ideally, this class should favor composition over inheritance. Since `DelegatingSecurityContextCallable` and `DelegatingSecurityContextRunnable` have factory methods that do what you need, would you be able to inline the support's `wrap` methods here in `DelegatingSecurityContextTaskScheduler`?

Reply:
wrap is not inlined in 1851a42